tests/Unit/Aggregate/AggregateRootWithAttributeApplyTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcing\Tests\Unit\Aggregate;
+
+use Patchlevel\EventSourcing\Aggregate\ApplyAttributeNotFound;
+use Patchlevel\EventSourcing\Tests\Unit\Fixture\Email;
+use Patchlevel\EventSourcing\Tests\Unit\Fixture\Message;
+use Patchlevel\EventSourcing\Tests\Unit\Fixture\MessageId;
+use Patchlevel\EventSourcing\Tests\Unit\Fixture\ProfileId;
+use Patchlevel\EventSourcing\Tests\Unit\Fixture\ProfileWithAttributeApply;
+use PHPUnit\Framework\TestCase;
+
+class AggregateRootWithAttributeApplyTest extends TestCase
+{
+    public function testApplyMethod(): void
+    {
+        $id = ProfileId::fromString('1');
+        $email = Email::fromString('david.badura@patchlevel.de');
+
+        $profile = ProfileWithAttributeApply::createProfile($id, $email);
+
+        self::assertEquals('1', $profile->aggregateRootId());
+        self::assertEquals(1, $profile->playhead());
+        self::assertEquals($id, $profile->id());
+        self::assertEquals($email, $profile->email());
+
+        $events = $profile->releaseEvents();
+
+        self::assertCount(1, $events);
+        $event = $events[0];
+        self::assertEquals(1, $event->playhead());
+    }
+
+    public function testMultipleApplyOnOneMethod(): void
+    {
+        $id = ProfileId::fromString('1');
+        $email = Email::fromString('david.badura@patchlevel.de');
+
+        $target = ProfileId::fromString('2');
+
+        $profile = ProfileWithAttributeApply::createProfile($id, $email);
+        $profile->visitProfile($target);
+
+        self::assertEquals('1', $profile->aggregateRootId());
+        self::assertEquals(1, $profile->playhead());
```

Comment:
why not `self::assertSame` for scalar values?

Reply:
We have already typesafety since the methods are declaring `string` or `int` as return value. But sure, `assertSame` should be always preferred i think.

Reply:
we can do this with help of reactor or php-cs-fixer over the whole codebase.